#!/bin/sh /usr/share/dpatch/dpatch-run
##
## 98_debian_alsaconf_nodup.dpatch by Thomas Hood
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Do not put any lines into /etc/mod{utils,probe.d}/sound that are
## DP: already present in /etc/mod{utils,probe.d}/alsa-base

@DPATCH@
diff -urNad alsa-utils-1.0.8+1.0.9rc2/alsaconf/alsaconf.in /tmp/dpep.CMl3CX/alsa-utils-1.0.8+1.0.9rc2/alsaconf/alsaconf.in
--- alsa-utils-1.0.8+1.0.9rc2/alsaconf/alsaconf.in	2005-03-25 15:05:41.041891852 +0100
+++ /tmp/dpep.CMl3CX/alsa-utils-1.0.8+1.0.9rc2/alsaconf/alsaconf.in	2005-03-25 15:08:01.589572000 +0100
@@ -649,6 +649,7 @@
 	    have_alias="yes"
 	fi
     fi
+if false ; then
     if [ -z "$have_alias" ]; then
 echo "alias char-major-116 snd
 alias char-major-14 $SOUND_CORE
@@ -666,16 +667,60 @@
     if [ -n "$CARD_OPTS" ]; then
 	echo "options $CARD_DRIVER $CARD_OPTS" >> $addcfg
     fi
+else
+    # For Debian
+    echo "alias snd-card-0 $CARD_DRIVER
+options $CARD_DRIVER index=0${CARD_OPTS:+ $CARD_OPTS}" >> $addcfg
+fi
 
     if [ -n "$cfgout" ]; then
 	[ ! -r "$cfgfile" ] || cmp -s "$TMP" "$cfgfile" || cat "$TMP" > "$cfgfile"
+if false ; then
 	cmp -s "$addcfg" "$cfgout" || cat "$addcfg" > "$cfgout"
+else
+	# For Debian
+	if [ "$kernel" = old ] ; then
+		if [ -f /etc/modutils/alsa-base ] ; then
+			# Don't duplicate any lines already in /etc/modutils/alsa-base
+			grep -Fv -f /etc/modutils/alsa-base "$addcfg" > "$TMP"
+		else
+			cat "$addcfg" > "$TMP"
+		fi
+	else
+		if [ -f /etc/modprobe.d/alsa-base ] ; then
+			# Don't duplicate any lines already in /etc/modprobe.d/alsa-base
+			grep -Fv -f /etc/modprobe.d/alsa-base "$addcfg" > "$TMP"
+		else
+			cat "$addcfg" > "$TMP"
+		fi
+	fi
+	cmp -s "$TMP" "$cfgout" || cat "$TMP" > "$cfgout"
+fi
     else
+if false ; then
 	echo "$ACB
 # --- ALSACONF version $version ---" >> $TMP
         cat "$addcfg" >> "$TMP"
 	echo "$ACE
 " >> $TMP
+else
+	# For Debian
+	if [ "$kernel" = old ] ; then
+		if [ -f /etc/modutils/alsa-base ] ; then
+			# Don't duplicate any lines already in /etc/modutils/alsa-base
+			grep -Fv -f /etc/modutils/alsa-base "$addcfg" >> "$TMP"
+		else
+			cat "$addcfg" >> "$TMP"
+		fi
+	else
+		if [ -f /etc/modprobe.d/alsa-base ] ; then
+			# Don't duplicate any lines already in /etc/modprobe.d/alsa-base
+			grep -Fv -f /etc/modprobe.d/alsa-base "$addcfg" >> "$TMP"
+		else
+			cat "$addcfg" >> "$TMP"
+		fi
+	fi
+fi
         cmp -s "$TMP" "$cfgfile" || cat "$TMP" > "$cfgfile"
     fi
 
